custom/di.go

```diff
@@ -45,6 +45,7 @@ type DeviceMfgInfo struct {
 	SerialNumber string
 	DeviceInfo   string
 	CertInfo     cbor.X509CertificateRequest
+	MACaddress   string
```

Comment:
While this fixes C compatibility, it breaks Java compatibility.

I suggest we resolve the C issue on the C side of things. See this issue: https://github.com/fido-device-onboard/client-sdk-fidoiot/issues/298

Reply:
The other option is to use a custom type `T` in `DIServer[T]` which has the 6th field.

You'd then just have to change two fields in the DI Server to use the new type.

```go
SignDeviceCertificate: yourCustomSignDeviceCertificate(state),
// This can stay the same
DeviceInfo: func(_ context.Context, info *YourCustomDeviceMfgInfo, _ []*x509.Certificate) (string, protocol.KeyType, protocol.KeyEncoding, error) {
	return info.DeviceInfo, info.KeyType, info.KeyEncoding, nil
},
```